src/guc.c

```diff
@@ -857,6 +933,19 @@ _guc_init(void)
 							   /* show_hook= */ NULL);
 #endif
 
+	DefineCustomStringVariable(MAKE_EXTOPTION("hyperstore_indexam_whitelist"),
+							   gettext_noop(
+								   "Whitelist for index access methods supported by hyperstore."),
+							   gettext_noop(
+								   "List of index access method names supported by hyperstore."),
+							   /* valueAddr= */ &ts_guc_hyperstore_indexam_whitelist,
+							   /* Value= */ "btree,hash",
+							   /* context= */ PGC_SIGHUP,
+							   /* flags= */ GUC_LIST_INPUT | GUC_SUPERUSER_ONLY,
+							   /* check_hook= */ check_indexam_whitelist,
+							   /* assign_hook= */ NULL,
+							   /* show_hook= */ NULL);
+
```

Comment:
Why is this a user-configurable option, the support for particular index types is hard-coded, right?

Reply:
It's not really hard-coded. We have have tests for these two index types, but there is nothing that says that it does not work for other index types, so it is possible to change in the configuration file.